### PR TITLE
added override for text writer

### PR DESCRIPTION
**Pull request by BrannonKing**
This is to improve performance as the JSON serializer uses the new override.

I was unable to test this with unit tests. I could not get the KVM stuff to run on my machine. Updating to the latest KVM made VS2015pre spin forever searching for tests.

### CODE REVIEW

**Comment by davidfowl** [comment]
I think we'll need at least one test for this. Also did you sign a CLA?

**Comment by BrannonKing** [comment]
I'm certainly open to writing a unit test. However, I couldn't see an immediate way to do one as it is Json.Net that calls the override. 

What is a CLA? I haven't signed anything.

**Comment by davidfowl** [comment]
@BrannonKing Can you resubmit the PR? A bot should appear telling you what you need to sign. This change looks great BTW

File: src/Microsoft.AspNet.SignalR.Server/Infrastructure/MemoryPoolTextWriter.cs [hunk 1]
```diff
@@ -144,6 +144,29 @@ public override void Write(string value)
...
+                    Encode(false);
+                }
+
+                var count = sourceLength - sourceIndex;
+                if (count > _textLength - _textEnd)
+                {
+                    count = _textLength - _textEnd;
+                }
+
+                Array.Copy(value, sourceIndex, _textArray, _textEnd, count);
```
Comment (davidfowl) [inline src/Microsoft.AspNet.SignalR.Server/Infrastructure/MemoryPoolTextWriter.cs]: value.CopyTo? To be consistent with the other method calls?

Comment (BrannonKing) [inline src/Microsoft.AspNet.SignalR.Server/Infrastructure/MemoryPoolTextWriter.cs]: Unfortunately, `value.CopyTo` doesn't have an overload that takes a source index.

**Comment by BrannonKing** [comment]
Closing pull request for resubmission.

**Pull request closed without merging**